backend/app.py

```diff
@@ -1,29 +1,41 @@
+# %%
 from jina.types.document.generators import from_csv
 from jina import DocumentArray, Flow
+
 from my_executors import ProtBertExecutor, MyIndexer
-from backend_config import protein_path
-from random import randint
+from backend_config import protein_path, embeddings_path
+from utils import load_or_download
+
+import os
 
+# %%
 
 def main():
-    proteins = DocumentArray(
-        from_csv(
-            fp=open(protein_path),
-            field_resolver={"Protein sequences": "text"},
+    # TODO: load the following from config file
+    url = "http://www.lri.fr/owncloud/index.php/s/fxIqHWvg1Zsq0JW/download"
+    pdb_data_path = protein_path #"../data/pdb_data_seq.csv"
+
+    with load_or_download(url, pdb_data_path) as data_file:
+        docs_generator = from_csv(
+            fp=data_file,
+            field_resolver={
+                "sequence": "text",
+                "structureId": "id"
+            }
         )
-    )
-    protein = proteins[randint(0, len(proteins)-1)]
-    print(f'Searching with protein sequence:\n{protein.text}')
+        proteins = DocumentArray(docs_generator)[0:42]
```

Comment:
What getting 42? Is this arbitrary to limit the computation time in development.

Or maybe it has something to do with li[fe, the Universe, and everything.](https://en.wikipedia.org/wiki/42_(number)#The_Hitchhiker's_Guide_to_the_Galaxy)

Reply:
It's definitely both ;)